torch/csrc/jit/runtime/symbolic_shape_registry.cpp

```diff
@@ -197,9 +304,31 @@ void registerSchema(
       graph->registerOutput(v);
     }
   }
-  // allow extra unused arguments to map multiple functions to e.g. unary
-  TORCH_INTERNAL_ASSERT(
-      graph->inputs().size() <= schema_string->arguments().size());
+}
+
+void registerSchema(
+    const FunctionSchema* schema_string,
+    const std::string& shape_compute_function_name,
+    std::unordered_map<std::string, std::shared_ptr<Graph>>& reused_functions,
+    const CompilationUnit& module) {
+  if (reused_functions.count(shape_compute_function_name)) {
+    auto graph = reused_functions[shape_compute_function_name];
+
+    // allow extra unused arguments to map multiple functions to e.g. unary
+    TORCH_INTERNAL_ASSERT(
+        graph->inputs().size() <= schema_string->arguments().size());
+
+    cached_schema_to_graph[schema_string] = graph;
+    return;
+  }
+
+  Function& shape_compute_function =
+      module.get_function(shape_compute_function_name);
+  std::shared_ptr<Graph> graph =
+      toGraphFunction(shape_compute_function).graph();
+
+  transformShapeFunction(schema_string, graph);
+  checkShapeFunction(schema_string, graph);
```

Comment:
Are all shape compute graphs hard coded? If so, can we move the checks to unit tests instead of doing it every time we load the module? Or perhaps I understand it incorrectly.